src/common/particleML.f90

```diff
@@ -68,4 +69,89 @@ module particleML
 !> will be allocated in allocatefieldsml::allocatefields
     type(particle), allocatable, public, save :: pdata(:)
 
+    contains
+      !> Scale activity of the particle,
+      !>
+      !> Activity remaining in the particle is
+      !> factor * original
+      !> The "lost" activity is returned
+      real function scale_rad(p, factor) result(removed)
+        class(particle), intent(inout) :: p
+        real, value :: factor
+        real :: previous
+
+        if (p%is_inactive()) then
+          removed = 0.0
+          return
+        endif
+
+        factor = min(factor, 1.0)
+        factor = max(factor, 0.0)
+
+        previous = p%rad_
+        p%rad_ = factor * previous
+        call p%flush_away_denormal()
+
+        removed = previous - p%rad_
+      end function
+
+      !> Set activity of particle
+      subroutine set_rad(p, rad)
+        class(particle), intent(inout) :: p
```

Comment:
Strange to see set_rad and add_rad as subroutine, while scale_rad is a function?

From a user perspective, I would like to have all subroutines here as functions.

Reply:
set_rad and add_rad should return final rad